src/HtmlUtils.js

```diff
@@ -519,7 +519,7 @@ export function bodyToHtml(content, highlights, opts={}) {
         emojiBody = match && match[0] && match[0].length === contentBodyTrimmed.length
                     // Prevent user pills expanding for users with only emoji in
                     // their username
-                    && !content.formatted_body.includes("https://matrix.to/");
+                    && (content.formatted_body && !content.formatted_body.includes("https://matrix.to/"));
```

Comment:
Actually I might be a liar, sorry. Should be !formatted || !includes because otherwise it'll give false positives

Reply:
Can you explain the logic here? With `||` it'll no longer check whether `formatted_body` is undefined and thus fail on the `includes` check if it is.

Did you mean:

```
... && (!content.formatted_body || (content.formatted_body && content.formatted_body.includes(...)))
```

?